examples/batch_dispatching.rs

```diff
@@ -0,0 +1,149 @@
+        .with_batch::<CustomBatchControllerSystem>(
+            DispatcherBuilder::new()
+                .with(BuyTomatoSystem, "buy_tomato_system", &[])
+                .with(BuyPotatoSystem, "buy_potato_system", &[]),
+            "BatchSystemTest",
+            &[],
+        )
+        .build();
+
+    let mut world = World::empty();
+
+    dispatcher.setup(&mut world);
+
+    // Running phase
+    for i in 0..10 {
+        println!("Dispatching {} ", i);
+
+        dispatcher.dispatch(&world);
+        sleep(Duration::new(0, 100000000));
+    }
+
+    // Done
+    println!("Execution finished");
+}
+
+// Resources
+
+#[derive(Default)]
+pub struct PotatoStore(i32);
+
+#[derive(Default)]
+pub struct TomatoStore(f32);
+
+// Say Hello Sytem
+
+pub struct SayHelloSystem;
+
+impl<'a> System<'a> for SayHelloSystem {
+    type SystemData = (Write<'a, PotatoStore>, Write<'a, TomatoStore>);
+
+    fn run(&mut self, _data: Self::SystemData) {
+        println!("Hello!")
+    }
+}
+
+// BuyPotatoSystem
+
+pub struct BuyPotatoSystem;
+
+impl<'a> System<'a> for BuyPotatoSystem {
+    type SystemData = (Write<'a, PotatoStore>);
+
+    fn run(&mut self, _data: Self::SystemData) {
+        println!("Buy Potato")
+    }
+}
+
+// Buy Tomato System
+
+pub struct BuyTomatoSystem;
+
+impl<'a> System<'a> for BuyTomatoSystem {
+    type SystemData = (Write<'a, PotatoStore>);
+
+    fn run(&mut self, _data: Self::SystemData) {
+        println!("Buy Tomato")
+    }
+}
+
+// TEST custom Batch Controller
+
+pub struct CustomBatchControllerSystem<'a, 'b> {
+    accessor: BatchAccessor,
+    dispatcher: Dispatcher<'a, 'b>,
+}
+
+impl<'a, 'b> BatchController<'a, 'b> for CustomBatchControllerSystem<'a, 'b> {
+    // Using this `BatchBuilderData = ()` make the dispatcher to panic since the run
+    // function will fetch the TomatoStore like the SayHelloSystem does.
+    // type BatchSystemData = ();
+    type BatchSystemData = (Read<'a, TomatoStore>);
+
+    fn create(accessor: BatchAccessor, dispatcher: Dispatcher<'a, 'b>) -> Self {
+        CustomBatchControllerSystem {
+            accessor,
+            dispatcher,
+        }
+    }
+}
+
+impl<'a> System<'a> for CustomBatchControllerSystem<'_, '_> {
+    type SystemData = BatchUncheckedWorld<'a>;
+
+    fn run(&mut self, data: Self::SystemData) {
+        {
+            // Unload before dispatch the inner dispatcher
+            let _ts = data.0.fetch::<TomatoStore>();
+        }
+        println!("Batch execution");
+        for _i in 0..3 {
+            self.dispatcher.dispatch(data.0);
+        }
+    }
+
+    fn running_time(&self) -> RunningTime {
+        RunningTime::VeryLong
+    }
+
+    fn accessor<'c>(&'c self) -> AccessorCow<'a, 'c, Self> {
+        AccessorCow::Ref(&self.accessor)
+    }
+
+    fn setup(&mut self, world: &mut World) {
+        self.dispatcher.setup(world);
+    }
+}
+
+unsafe impl<'a, 'b> Send for CustomBatchControllerSystem<'a, 'b> {}
```

Comment:
I actually missed this `impl` first time reading through. I would highlight it by adding comment explaining it.